kubeflow/argo/argo.libsonnet

```diff
@@ -0,0 +1,339 @@
+                  "workflow-controller",
+                ],
+                env: [
+                  {
+                    name: "ARGO_NAMESPACE",
+                    valueFrom: {
+                      fieldRef: {
+                        apiVersion: "v1",
+                        fieldPath: "metadata.namespace",
+                      },
+                    },
+                  },
+                ],
+                image: "argoproj/workflow-controller:v2.0.0-beta1",
+                imagePullPolicy: "IfNotPresent",
+                name: "workflow-controller",
+                resources: {},
+                terminationMessagePath: "/dev/termination-log",
+                terminationMessagePolicy: "File",
+              },
+            ],
+            dnsPolicy: "ClusterFirst",
+            restartPolicy: "Always",
+            schedulerName: "default-scheduler",
+            securityContext: {},
+            serviceAccount: "argo",
+            serviceAccountName: "argo",
+            terminationGracePeriodSeconds: 30,
+          },
+        },
+      },
+    },  // deploy
+
+
+    deployUi: {
+      apiVersion: "extensions/v1beta1",
+      kind: "Deployment",
+      metadata: {
+        labels: {
+          app: "argo-ui",
+        },
+        name: "argo-ui",
+        namespace: namespace,
+      },
+      spec: {
+        progressDeadlineSeconds: 600,
+        replicas: 1,
+        revisionHistoryLimit: 10,
+        selector: {
+          matchLabels: {
+            app: "argo-ui",
+          },
+        },
+        strategy: {
+          rollingUpdate: {
+            maxSurge: "25%",
+            maxUnavailable: "25%",
+          },
+          type: "RollingUpdate",
+        },
+        template: {
+          metadata: {
+            creationTimestamp: null,
+            labels: {
+              app: "argo-ui",
+            },
+          },
+          spec: {
+            containers: [
+              {
+                env: [
+                  {
+                    name: "ARGO_NAMESPACE",
+                    valueFrom: {
+                      fieldRef: {
+                        apiVersion: "v1",
+                        fieldPath: "metadata.namespace",
+                      },
+                    },
+                  },
+                  {
+                    name: "IN_CLUSTER",
+                    value: "true",
+                  },
+                ],
+                image: "argoproj/argoui:v2.0.0-beta1",
+                imagePullPolicy: "IfNotPresent",
+                name: "argo-ui",
+                resources: {},
+                terminationMessagePath: "/dev/termination-log",
+                terminationMessagePolicy: "File",
+              },
+            ],
+            dnsPolicy: "ClusterFirst",
+            restartPolicy: "Always",
+            schedulerName: "default-scheduler",
+            securityContext: {},
+            serviceAccount: "argo",
+            serviceAccountName: "argo",
```

Comment:
Oh I forgot to mention another change is that there are now two service accounts: one for controller, the other for UI with different permissions. (e.g. the UI needs access to logs whereas the controller does not). I notice in this change the UI is using the same service account (argo). I think you will want to create the argo-ui service account mapped to these permissions:

https://github.com/argoproj/argo/blob/master/cmd/argo/commands/const.go#L44

Reply:
I think you did mention that. PTAL.